compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/BIRPackageSymbolEnter.java

```diff
@@ -255,10 +255,42 @@ private BPackageSymbol definePackage(DataInputStream dataInStream, int pkgCpInde
         // Define service declarations
         defineSymbols(dataInStream, rethrow(this::defineServiceDeclarations));
 
+        populateReferencedFunctions();
+
         this.typeReader = null;
         return this.env.pkgSymbol;
     }
 
+    private void populateReferencedFunctions() {
+        for (BStructureTypeSymbol structureTypeSymbol : this.structureTypes) {
+            if (structureTypeSymbol instanceof BObjectTypeSymbol) {
```

Comment:
Can't/shouldn't we check `structureTypeSymbol.type.tag` instead? Or at least symbol kind.

Reply:
Updated in 1b6b21ff9a8ad716b114fd76ad04f19dc70a57e8